src/openrct2/windows/new_ride.c

```diff
@@ -517,11 +517,18 @@ void window_new_ride_focus(ride_list_item rideItem)
 
 	rideType = get_ride_entry(rideItem.entry_index);
 
-	if(!gConfigInterface.select_by_track_type)
+	if(!gConfigInterface.select_by_track_type) {
 		window_new_ride_set_page(w, rideType->category[0]);
-	else
-		window_new_ride_set_page(w, gRideCategories[rideType->ride_type[0]]);
-
+	}
+	else {
+		// Find the first non-null ride type index.
+		uint8 rideTypeIndex = 0;
+		while(rideType->ride_type[rideTypeIndex] == RIDE_TYPE_NULL && rideTypeIndex < 3) {
+			rideTypeIndex++;
+		}
+		window_new_ride_set_page(w, gRideCategories[rideTypeIndex]);
+	}
+	
```

Comment:
In the next break, check if the search has turned up anything, if not, select the first entry that has the same ride group.